chronosphere/generateresources/main.go

```diff
@@ -85,35 +88,53 @@ type entityType struct {
 	SwaggerType string
 	// SwaggerModel represents the underlying model to use. Will normally be the
 	// same as SwaggerType
-	SwaggerModel         string
-	FieldName            string
-	SwaggerClient        string
-	SwaggerClientPackage string
-	DisableDryRun        bool
-	UpdateUnsupported    bool
-	SingletonID          string
-	SingletonIDConst     string
+	SwaggerModel          string
+	FieldName             string
+	SwaggerClient         string
+	SwaggerClientPackage  string
+	DisableDryRun         bool
+	UpdateUnsupported     bool
+	SingletonID           string
+	SingletonIDConst      string
+	SlugField             string
+	EntityLinkedSingleton bool
 }
 
 func newEntityType(a api, r registry.Resource) entityType {
+	var (
+		slugField             = defaultSlugField
+		entityLinkedSingleton bool
+	)
+	if r.EntityLinkedSingletonSlugField != "" {
+		slugField = r.EntityLinkedSingletonSlugField
+		entityLinkedSingleton = true
+	}
 	et := entityType{
-		API:                  a,
-		GoType:               fmt.Sprintf("%s%s", a.GoPrefix, r.Entity),
-		SwaggerType:          r.Entity,
-		SwaggerModel:         r.Entity,
-		SwaggerClient:        fmt.Sprintf("%s.%s", a.Client, r.Entity),
-		SwaggerClientPackage: strcase.ToSnake(r.Entity),
-		DisableDryRun:        r.DisableDryRun,
-		UpdateUnsupported:    r.UpdateUnsupported,
-		SingletonID:          r.SingletonID,
-		SingletonIDConst:     fmt.Sprintf("%sID", r.Entity),
+		API:                   a,
+		GoType:                fmt.Sprintf("%s%s", a.GoPrefix, r.Entity),
+		SwaggerType:           r.Entity,
+		SwaggerModel:          r.Entity,
+		SwaggerClient:         fmt.Sprintf("%s.%s", a.Client, r.Entity),
+		SwaggerClientPackage:  strcase.ToSnake(r.Entity),
+		DisableDryRun:         r.DisableDryRun,
+		UpdateUnsupported:     r.UpdateUnsupported,
+		SingletonID:           r.SingletonID,
+		SingletonIDConst:      fmt.Sprintf("%sID", r.Entity),
+		SlugField:             slugField,
+		EntityLinkedSingleton: entityLinkedSingleton,
 	}
 
 	// special case for classic dashboards
 	if r.Name == "classic_dashboard" {
 		et.GoType = fmt.Sprintf("%s%s", a.GoPrefix, "ClassicDashboard")
 	}
 
+	// special case for service attributes - client is in config_v1 package
```

Comment:
do you know why it's different for service_attributes? maybe worth a comment with context for why